builtin/providers/openstack/resource_openstack_compute_instance_v2.go

```diff
@@ -965,18 +965,19 @@ func getInstanceNetworks(computeClient *gophercloud.ServiceClient, d *schema.Res
 		allPages, err := tenantnetworks.List(computeClient).AllPages()
 		if err != nil {
 			if _, ok := err.(gophercloud.ErrDefault404); ok {
+				log.Println("[DEBUG] os-tenant-networks disabled")
 				tenantNetworkExt = false
 			}
 
+			log.Println("[DEBUG] Err looks like: %+v", err)
 			if errCode, ok := err.(gophercloud.ErrUnexpectedResponseCode); ok {
```

Comment:
Actually, thinking about it I don't think this will ever be 'ok', as the 'AllPages()' function strips away all error context, and instead you're just left with a string. Line 967 works, as it's attempting to assert said string to 'gophercloud.ErrDefault404'